fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/synonym/SynonymMapConfig.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SynonymMapConfig.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.synonym;
+
+import com.apple.foundationdb.record.RecordCoreException;
+import com.apple.foundationdb.record.logging.LogMessageKeys;
+
+import javax.annotation.Nonnull;
+import java.io.InputStream;
+
+public interface SynonymMapConfig {
+    /**
+     * Get the name of the synonymMap to build
+     * @return the name
+     */
+    String getName();
+
+    /**
+     * Get solr format input stream to build the synonymMap
+     * @return the input stream
+     */
+    InputStream getSynonymInputStream();
```

Comment:
It seems strange to me that this enforces using a specific storage format of the synonyms, rather than having the interface that gets registered return a `SynonymMap` directly.

Reply:
I think this is probably fine for now, but I'm also a little hesitant about using an `InputStream` here, as it effectively means that the solr file format becomes a part of our API, rather than using something more neutral. This class isn't _technically_ marked as `EXPERIMENTAL` (though maybe we should fix that), but since it's all part of this experimental lucene index stuff, if we come up with a better way of expressing it.

Reply:
I marked the class as EXPERIMENTAL for now until we figure this out.